rust/meson.build

```diff
@@ -1 +1,2 @@
 subdir('scx_utils')
+subdir('scx_user')
```

Comment:
I'm glad that you're saying this. In my initial version I actually used `scx_rustland` for this crate, but I wasn't 100% convinced about reusing the name for something else... Maybe `scx_rustland` should just evolve into this meta-scheduler crate that allows to write schedulers in Rust?

And then we can unlock a new namespace that would even allow to potentially re-implement the same scheduling policies that we have in C / BPF, without conflicting with the naming.

Reply:
I've updated this PR (only locally for now) to rename `scx_user` -> `scx_rustland`, `scx_fifo` -> `scxrl_fifo` and `scx_rustland` -> `scxrl_vtime`.

However, I still have conflicting opinions about this renaming.

I don't see any problem with user and fifo, but about rustland, we have docs and press material (blog posts, articles, etc.) that link the `scx_rustland` directory in this repo directly. We are going to break the existing scx_rustland crate (or at least change its nature, from being a scheduler to a framework). And we also have the `scx` systemd service that has the name `scx_rustland`.

If we rename the scheduler we're going to break all of the above potentially.

So, I was considering the following options:
 1. rename everything as planned, keep `scheds/rust/scx_rustland/README.md` to explain the renaming, add some logic to the scx service script to print a warning if someone is trying to use scx_rustland as a scheduler (not sure how to handle the change in the crate in this case)
 1. rename `scx_user` -> `scx_rustland_core`, keep `scx_rustland` as it is, rename `scx_fifo` -> `scx_rlfifo`

I'm more inclined to go with option 2 (at the end `scx_rustland_core` is not a bad name for a user-space Rust scheduling framework) and we have the benefits of not breaking anything in the existing docs, articles, links, tools, etc.

Opinions?

Reply:
ack, in the meantime I've pushed an update to go with option 2. In this way we're not breaking anything and we're not allocating `scx_fifo` (the "rustland FIFO" example is now called `scx_rlfifo`).

I'd like to see if I can reduce the code duplication between `scx_utils` and `scx_rustland_core` and then I think we may be ready to merge this and keep improving this stuff in tree.

Reply:
...and with the last push I think I've addressed/fixed all the other comments, so the only thing that I'd like to look at before merging this PR is the `BpfUserBuilder` wrapper.

Reply:
Alright, with 3b4df81913e4a474db79789a4b1b32e81a7c771e on top I think we have a fairly reasonable first implementation of the `scx_rustland_core` framework.

Thinking more about shpping the pre-compiled BPF component of `scx_rustland_core` in the crate, I think I actually prefer to embed `main.bpf.c` as source code in the crate (as it is right now), because in this way it'll be always compiled with the same toolchain that is used for the scheduler, that seems more robust.